uri/uri_test.cpp

```diff
@@ -178,5 +182,34 @@ int main() {
         expect_eq(completed, uri::Uri::parse("hax://example.com/test/hello"));
     });
 
+    etest::test("blob URL generation", [] {
+        std::string REGEX_UUID = "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}";
+
+        uri::Host h = {uri::DNS_DOMAIN, "example.com"};
+        uri::Origin o = {"https", &h, (std::uint16_t)8080, std::nullopt, false};
+
+        std::string blob = uri::blob_url_create(&o);
+        std::cout << "Generated Blob URL: " << blob << std::endl;
+
+        etest::require(std::regex_match(blob, std::regex("blob:https://example.com:8080/" + REGEX_UUID)));
```

Comment:
In general, use `expect` when you don't require the test to bail on this line if this assertion fails.